sources/lib/DependencyInjection/Compiler/ModelPass.php

```diff
@@ -49,23 +52,45 @@ private function addTagged(DI\ContainerBuilder $container, $tag, $defaultService
                     if (!in_array($interface, class_implements($definitions[$serviceId]->getClass()), true)) {
                         throw new \RuntimeException(sprintf('Your pooler should implement %s.', $interface));
                     }
+
+                    $poolerToService[$serviceId] = [];
                 } else {
                     throw new \RuntimeException(sprintf('There is no pooler service with id %s.', $serviceId));
                 }
             }
 
-            $definitions[$serviceId]->addMethodCall('addModelToServiceMapping', [$class, $id . '.pomm.inner']);
+            $innerId = $id . '.pomm.inner';
+            $definitions[$serviceId]->addMethodCall('addModelToServiceMapping', [$class, $innerId]);
+            $poolerToService[$serviceId][$innerId] = new DI\Reference($innerId);
 
             $old = $container->getDefinition($id);
             $old->setPublic(true);
             $container->removeDefinition($id);
-            $container->addDefinitions([$id . '.pomm.inner' => $old]);
+            $container->addDefinitions([$innerId => $old]);
 
-            $container->register($id, $old->getClass())
+            $service = $container->register($id, $old->getClass())
                 ->setFactory([new DI\Reference($sessionId), $method])
                 ->addArgument($old->getClass())
-                ->setAutowiringTypes([$old->getClass()]) //set this one as a default for autowire
             ;
+
+            if ($gteSf33) {
```

Comment:
I think it’s a good idea to split this method into two functions to reduce complexity and length of this one.